### PR TITLE
Tabbar icon

### PR DESCRIPTION
**Pull request by ayanshakil**
- Added a Bar indicator below the selected icons in tab bar
- removed the background ripple, added fade in fade out animation to tinted icon
- Added rounded corners to the indicator and horizontal shrink and expand animation to the bar.

### CODE REVIEW

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt [hunk 1]
```diff
@@ -64,20 +68,22 @@ fun TabBar(
...
-                    icon = if (tabData.selected) tabData.selectedIcon else tabData.icon,
-                    textAlignment = tabTextAlignment,
-                    selected = tabData.selected,
-                    onClick = tabData.onClick,
-                    accessory = tabData.badge,
-                    tabItemTokens = tabItemTokens
-                        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.TabItemControlType] as TabItemTokens,
-                    style = if (tabData.selected) FluentStyle.Brand else FluentStyle.Neutral
-                )
+
```
Comment (praveen970) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt:71]: Looks like there are no changes. Please check the formatting

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt [hunk 2]
```diff
@@ -1,10 +1,14 @@
 package com.microsoft.fluentui.tokenized.navigation
 
 import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
```
Comment (praveen970) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt:4]: Remove unused imports

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 3]
```diff
@@ -23,11 +35,14 @@ import androidx.constraintlayout.compose.ConstraintLayout
 import androidx.constraintlayout.compose.Dimension
 import com.microsoft.fluentui.theme.FluentTheme
 import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.FluentAliasTokens
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:38]: Check for unused imports

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 4]
```diff
@@ -209,6 +225,22 @@ fun TabItem(
...
+                    visible = selected,
+                    enter = fadeIn(animationSpec = tween(durationMillis = 300))+ expandHorizontally(animationSpec = tween(durationMillis = 300)),
+                    exit = fadeOut(animationSpec = tween(durationMillis = 300))+ shrinkHorizontally(animationSpec = tween(durationMillis = 300)),
+                )
+                {
+                    Box(
+                        modifier = Modifier
+                            .height(2.dp)
+                            .width(FluentGlobalTokens.SizeTokens.Size160.value)
+                            .background(shape = RoundedCornerShape(FluentGlobalTokens.CornerRadiusTokens.CornerRadiusCircle.value), color = textColor)
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:240]: Move fluentGlobaltokens into a variable and use here

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt [hunk 5]
```diff
@@ -63,14 +63,15 @@ open class TabItemTokens : IControlToken, Parcelable {
...
+
     @Composable
     open fun iconColor(tabItemInfo: TabItemInfo): StateColor {
         return when (tabItemInfo.fluentStyle) {
             FluentStyle.Neutral -> StateColor(
                 rest = FluentTheme.aliasTokens.neutralForegroundColor[FluentAliasTokens.NeutralForegroundColorTokens.Foreground3].value(
                     themeMode = FluentTheme.themeMode
                 ),
-                pressed = FluentTheme.aliasTokens.neutralForegroundColor[FluentAliasTokens.NeutralForegroundColorTokens.Foreground3].value(
+                pressed = FluentTheme.aliasTokens.brandForegroundColor[FluentAliasTokens.BrandForegroundColorTokens.BrandForeground1].value(
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt:74]: Should we also consider changing text color as well in pressed state?

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 6]
```diff
@@ -209,6 +225,22 @@ fun TabItem(
...
+
+                AnimatedVisibility(
+                    visible = selected,
+                    enter = fadeIn(animationSpec = tween(durationMillis = 300))+ expandHorizontally(animationSpec = tween(durationMillis = 300)),
+                    exit = fadeOut(animationSpec = tween(durationMillis = 300))+ shrinkHorizontally(animationSpec = tween(durationMillis = 300)),
+                )
+                {
+                    Box(
+                        modifier = Modifier
+                            .height(2.dp)
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:238]: should be 3.dp

Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt]: corrected it to 3.dp

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt (hunk 5 above)
Comment (ayanshakil) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt:73]: changed the text color too in the pressed state.

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt (hunk 4 above)
Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt]: created the variable for it , done.

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt (hunk 3 above)
Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:35]: removed them

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt (hunk 2 above)
Comment (ayanshakil) [reply fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt]: removed the unused imports

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt (hunk 1 above)
Comment (ayanshakil) [reply fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt]: yes changed it

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt [hunk 7]
```diff
@@ -91,7 +91,7 @@ open class TabItemTokens : IControlToken, Parcelable {
                     )
                 ).value(FluentTheme.themeMode),
                 pressed = FluentColor(
-                    light = FluentTheme.aliasTokens.brandForegroundColor[FluentAliasTokens.BrandForegroundColorTokens.BrandForeground1Pressed].value(
+                    light = FluentTheme.aliasTokens.brandForegroundColor[FluentAliasTokens.BrandForegroundColorTokens.BrandForeground1].value(
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt:94]: This is FluentStyle.Brand. This should be BeandForground1Pressed

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 8]
```diff
@@ -91,12 +106,11 @@ fun TabItem(
             tint = iconColor
         )
     }
-
     if (textAlignment == TabTextAlignment.HORIZONTAL) {
         ConstraintLayout(
             modifier = modifier
                 .then(clickableModifier)
-                .background(backgroundColor)
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:99]: Why is this removed?

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 9]
```diff
@@ -191,13 +205,13 @@ fun TabItem(
...
         }
-
+        val cornerRadiusSize = FluentGlobalTokens.CornerRadiusTokens.CornerRadiusCircle.value
+        val indicatorWidth = FluentGlobalTokens.SizeTokens.Size160.value
         Column(
             horizontalAlignment = Alignment.CenterHorizontally,
             verticalArrangement = Arrangement.Center,
             modifier = modifier
                 .then(clickableModifier)
-                .background(backgroundColor)
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:200]: Why is this removed?

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 10]
```diff
@@ -191,13 +205,13 @@ fun TabItem(
                 }
             }
         }
-
+        val cornerRadiusSize = FluentGlobalTokens.CornerRadiusTokens.CornerRadiusCircle.value
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:208]: rename the variable to indicatorCornerRadiusSize and same for other variables

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt (hunk 9 above)
Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:200]: to remove the background black shadow that appears on clicking the icon

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt [hunk 11]
```diff
@@ -64,20 +64,20 @@ fun TabBar(
...
-                    icon = if (tabData.selected) tabData.selectedIcon else tabData.icon,
-                    textAlignment = tabTextAlignment,
-                    selected = tabData.selected,
-                    onClick = tabData.onClick,
-                    accessory = tabData.badge,
-                    tabItemTokens = tabItemTokens
-                        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.TabItemControlType] as TabItemTokens,
-                    style = if (tabData.selected) FluentStyle.Brand else FluentStyle.Neutral
-                )
+                    TabItem(
```
Comment (praveen970) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tokenized/navigation/TabBar.kt:67]: Restore the file if no changes are present. not required to be included in this PR

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt [hunk 12]
```diff
@@ -63,14 +62,15 @@ open class TabItemTokens : IControlToken, Parcelable {
         )
     }
 
+
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt:65]: Remove this space

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 13]
```diff
@@ -23,6 +32,8 @@ import androidx.constraintlayout.compose.ConstraintLayout
 import androidx.constraintlayout.compose.Dimension
 import com.microsoft.fluentui.theme.FluentTheme
 import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.FluentAliasTokens
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:35]: is FluentAliasTokens needed here?

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt (hunk 8 above)
Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:99]: to remove the background shadow that appears when clicked or pressed

**Comment by praveen970** [comment]
Can you add screenshot with after and before changes

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt (hunk 12 above)
Comment (ayanshakil) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/TabItemTokens.kt]: done

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt (hunk 13 above)
Comment (ayanshakil) [reply fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt]: no not needed , removed it

**Comment by ayanshakil** [comment]
Screenshot_20240623-223139
Screenshot_20240623-225335

File: fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt [hunk 14]
```diff
@@ -192,12 +212,13 @@ fun TabItem(
...
         }
 
+        val indicatorCornerRadiusSize = FluentGlobalTokens.CornerRadiusTokens.CornerRadiusCircle.value
+        val indicatorWidth = FluentGlobalTokens.SizeTokens.Size160.value
         Column(
             horizontalAlignment = Alignment.CenterHorizontally,
             verticalArrangement = Arrangement.Center,
             modifier = modifier
                 .then(clickableModifier)
-                .background(backgroundColor)
```
Comment (praveen970) [inline fluentui_listitem/src/main/java/com/microsoft/fluentui/tokenized/tabItem/TabItem.kt:200]: instead of removing the background and indication add a param "showHandle" and based on this flag handle the backgrounf and indication
